chitchat/src/state.rs

```diff
@@ -137,12 +138,9 @@ impl NodeState {
         self.max_version = max_version;
     }
 
-    // Prepare the node state to receive a delta.
-    // Returns `true` if the delta can be applied. In that case, the node state may be mutated (if a
-    // reset is required) Returns `false` if the delta cannot be applied. In that case, the node
-    // state is not modified.
+    // Assess whether the delta can be applied or not.
     #[must_use]
-    fn prepare_apply_delta(&mut self, node_delta: &NodeDelta) -> bool {
+    fn check_delta_status(&self, node_delta: &NodeDelta) -> DeltaStatus {
```

Comment:
I just isolated the immutable logic that considers whether we should accept a delta or not.

I also tried to make it simpler.